src/Common/Services/EnvironmentDeploymentHandler.cs

```diff
@@ -24,6 +24,11 @@ public class EnvironmentDeploymentHandler : IEnvironmentDeploymentHandler
         private bool _isDeploymentFinished = false;
         private bool _hasDeploymentFailed = false;
 
+        private static DateTime? utcOfLastMessageReceived = null;
+
+        private TimeSpan timeoutMainTask = TimeSpan.FromMinutes(60);
+        private TimeSpan timeoutToGetSomeMBMessageTask = TimeSpan.FromMinutes(15);

```

Comment:
Sorry to meddle, but I still get notifications on this repo, and thought I could help.

From experience, this kind of timeout should have a default value but also be configurable (using an env var or command line parameter).
Most likely it will never be used but there will be a day when someone will find a place where it will be needed to increase the timeout, especially `timeoutMainTask`, and you wish this was configurable 😄